src/mvp/main.cpp

```diff
@@ -107,17 +107,30 @@ void CreateStartMenu() {
     UI::menu::ClickOption connectOption("connectOption", 1.0,
                                         "Connect to server");
     connectOption.onClick.Subscribe(
-        [](void *, events::MouseClickEventArgs &args) {
+        [gameManager](void *, events::MouseClickEventArgs &args) {
             if (args.pressed) {
-                GameData::engine->SetAndGetDisplayLayout(g_dlStore);
-                GameData::engine->SetAndGetScene(g_sStore);
+                bool connected = gameManager->ConnectToServer();
+                if (connected) {
+                    GameData::engine->SetAndGetDisplayLayout(g_dlStore);
+                    GameData::engine->SetAndGetScene(g_sStore);
```

Comment:
If we assume we are never going back to the start menu again this is reasonable. Otherwise we should update the current store with the old value, like:

```c++
g_dlStore = GameData::engine->SetAndGetDisplayLayout(GameData::startMenu);
g_sStore = GameData::engine->SetAndGetScene(GameData::startMenuScene);
```

Reply:
We didn't think about this earlier but now is a good time to change this.